src/eolab/rastertools/processing/sliding.py

```diff
@@ -56,8 +56,12 @@ def compute_sliding(input_image: str, output_image: str, rasterprocessing: Raste
             if src.height < blockysize:
                 blockysize = utils.highest_power_of_2(src.height)
 
-            # dtype of output data
-            dtype = rasterprocessing.dtype or rasterio.float32
+            # dtype and creation options of output data
+            dtype_algo = np.float32 or rasterio.float32
```

Comment:
I defined there a specific float32 dtype because a int8 dtype is not working at 
`processing/algo.py, line 511 : new_ratios /= (r * resolution)` -> if new_ratios is int8 dtypes, there is a casting error `numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'divide' output from dtype('float64') to dtype('uint8') with casting rule 'same_kind'`.
So `dtype = np.ubyte` is used as the data type of the hillshade product, and `dtype_algo = np.float32` is used as the data type for hillshade computation.


Another workaround would be to : 
- Replace add `processing/algo.py, line 510 : new_ratios = new_ratios.astype(np.float32)` just before the division line 511
- Modify `processing/sliding.py, line 124 : dataset = dataset.astype(np.float32)`.
- Remove the argument `dtype` from `processing/sliding.py: _process_sliding`


In both cases, the hillshade output is exactly what we want and the output size is lighter than before

Reply:
I'm not sure where the new_ratios is casted to int8? What would be better would be to just force the dtypes into the algo file, but not on line 511, I would do it here:  https://github.com/CNES/rastertools/blob/main/src/eolab/rastertools/processing/algo.py#L499. But I'm not sure why input_bands is int8. And I'm a bit afraid this change cause trouble in case of uint8 and negative values.

This also makes me see that out dtype should be bool or something else than float: https://github.com/CNES/rastertools/blob/main/src/eolab/rastertools/processing/algo.py#L496.

Reply:
**About the int8 cast :**
- If I define an unique dtype to np.ubyte (which is actually a np.uint8 dtype), then at `processing/sliding.py:L110 : dataset = dataset.astype(dtype)` and so dataset is casted to np.uint8.
> First issue : dataset initially contains floating values of elevation (DEM is the input). So by casting it to int8 or uint8 we degrade valuable data.
-  Then at L128 `output = rasterprocessing.compute(dataset)` and so there we are in `processing/algo.py:L474 at hillshade` where `new_ratios` is dtyped as `input_band` or `input_data`. So `new_ratios` is dtyped np.uint8.
Then we have `processing/algo.py, line 511 : new_ratios /= (r * resolution)`, so 
`new_ratios` (dtype=np.uint8) = ` [new_ratios / float value]` (dtype = np.float64). 
> There is the casting issue.


**About setting `out = np.zeros(shape, dtype=bool)`:**
- It is indeed possible to set `out` dtype to bool but when `out` will be written in the output file at `processing/sliding.py:L131 at _process_sliding` , the dtype of dst will be recasted as it was initialized (because of "r+" mode). And so final output will remain an np.int8 or np.uint8, and so the output final size will not change. So I am not sure if it is useful but it looks harmless anyway.

Reply:
Okay, I understand better. The problem with the current modification is that it could affect other algorithms, as it is not specific to hillshade algorithm. Any idea to avoid that?

About the use of a boolean output: it would optimize memory consuption and probably computing speed a bit (but this is also almost certainly negligible).

Reply:
Well as I much as I can see the only modification that can affect other algorithms is `dtype_algo=np.float32 or rasterio.float32` instead of `dtype=rasterprocessing.dtype or rasterio.float32` in `processing/sliding.py:L60` because dtype is defined in hillshade.py, svf.py and filtering.py. It means there will be a difference if and only if rasterprocessing.dtype was different from np.float32.

I checked the rasterprocessing dtype set in svf.py -> dtype = np.float32 ; in filtering.py -> dtype is defined as the constructor value of RasterFilter -> dtype = np.float32

Reply:
Yes exactly, but it should remain configurable if an algo needs a different one. So I think this specific change cannot be made.

Reply:
The only solution I see is to have both an out_dtype and a processing_dtype. But his means modifying all other algorithms. Maybe we could keep `dtype`and `processing_dtype` (or `algo_dtype` as you used). But both should be rasterprocessing argument. In this case:
```
dtype = rasterprocessing.dtype or rasterio.float32
processing_dtype = rasterprocessing.processing_dtype or dtype 
```

This way, we add a feature without having to modify other algorithms classes.

@queyruto what do you thin?

Reply:
It sounds good to me